frontend/src/components/TicketList.js

```diff
@@ -25,38 +26,54 @@ const TicketList = (props) => {
         .flat()
         .filter((ticket) => filter.includes(ticket.type))
 
+    const isAllTicketsEmpty =
+        Object.keys(TICKET_TYPES)
+            .map((type) => allTickets[type])
+            .flat().length === 0
+
     return (
         <Box minW={{ base: '200px', lg: '280px', xl: '300px' }}>
             <FilterDropdown filter={filter} setFilter={setFilter} />
-            <VStack
-                pos="relative"
-                w="100%"
-                h="calc(100vh - 152px)"
-                overflowY="scroll"
-            >
-                {filteredTickets.map((ticket) => {
-                    const code = `${ticket.type}-${ticket.id}`
-                    return (
-                        <Card
-                            key={code}
-                            w="100%"
-                            borderTop="1px solid #dedede"
-                            borderBottom="1px solid #dedede"
-                            borderRadius="0"
-                            mt="0 !important"
-                            onClick={() =>
-                                navigate(`/${ticket.type}/${ticket.id}`)
-                            }
-                            cursor="pointer"
-                        >
-                            <CardBody p="8px 16px">
-                                <Text>{code}</Text>
-                                <Text noOfLines="1">{ticket.name}</Text>
-                            </CardBody>
-                        </Card>
-                    )
-                })}
-            </VStack>
+
+            {isAllTicketsEmpty ? (
```

Comment:
would prefer in this case to just return spinner early as opposed a ternary as the code is quite long